runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/environment/ProcessEnvironmentFactory.java

```diff
@@ -104,29 +105,37 @@ private ProcessEnvironmentFactory(
   /** Creates a new, active {@link RemoteEnvironment} backed by a forked process. */
   @Override
   public RemoteEnvironment createEnvironment(Environment environment) throws Exception {
-    String workerId = idGenerator.getId();
+    Preconditions.checkState(
+        environment
+            .getUrn()
+            .equals(RunnerApi.StandardEnvironments.Environments.EXTERNAL.toString()),
+        "The passed environment does not contain a ProcessPayload.");
+    final RunnerApi.ProcessPayload processPayload =
+        RunnerApi.ProcessPayload.parseFrom(environment.getPayload());
+    final String workerId = idGenerator.getId();
 
-    // TODO The Environment Protobuf message needs to be changed for process environment
-    String executable = environment.getUrl();
+    String executable = processPayload.getCommand();
+    ImmutableList.Builder<String> args =
+        ImmutableList.<String>builder().addAll(processPayload.getArgsList());
```

Comment:
This prepends the process arguments from the proto before the container contract arguments. I would prefer if parameters were only passed via environment variables, to separate the containers arguments from other arguments.

We can still make adjustments later on, so this looks good for now. If the args are empty, it doesn't effect the current behavior.